state-chain/runtime/src/chainflip.rs

```diff
@@ -626,3 +628,27 @@ pub fn calculate_account_apy(account_id: &AccountId) -> Option<u32> {
 			.unwrap_or_default()
 	})
 }
+
+/// Types used for subscription
+pub trait WithContent<Data> {
+	fn content(&self) -> Data;
+}
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, Eq, Debug, Encode, Decode)]
+pub struct SubscriptionUpdate<Data> {
+	pub block_hash: Hash,
+	pub block_number: BlockNumber,
+	pub data: Data,
```

Comment:
You should use serde(flatten) here, so the "data" doesn't show up in all the rpc responses. Also you need to create a OrderFills structure to use as Data instead of just a Vec<>. Again to preserve the response structure (For back compat).

Reply:
The OrderFilled is not contained in a struct. 
I run the test app to ensure the result is indeed flattened